src/calculate.py

```diff
@@ -39,36 +38,55 @@ def compute_yearly(self, year):
                     highest_humidity = reading.humidity
                     highest_humidity_date = reading.date
 
-        return YearlyResult(
-            highest_temp,
-            highest_temp_date,
-            lowest_temp,
-            lowest_temp_date,
-            highest_humidity,
-            highest_humidity_date,
-        )
+        highest_temp_date_str = self.format_date(highest_temp_date)
+        lowest_temp_date_str = self.format_date(lowest_temp_date)
+        highest_humidity_date_str = self.format_date(highest_humidity_date)
+
+        result = {
+            "highest_temp": f"Highest: {highest_temp}C on {highest_temp_date_str}",
+            "lowest_temp": f"Lowest: {lowest_temp}C on {lowest_temp_date_str}",
+            "highest_humidity": f"Humidity: {highest_humidity}% on {highest_humidity_date_str}",
+        }
 
-    def compute_monthly(self, month):
-        """_summary_
+        return result
+
+    def print_yearly_report(self, result):
+        [print(value) for value in result.values()]
+
+    def format_date(self, temp_date):
```

Comment:
this should not be in this specific class.
It should be a helper function in a separate file